File: app/src/main/java/com/example/dcct/ui/fragment/LoginFragment.java
package com.example.dcct.ui.fragment;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;


import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.dcct.R;
import com.example.dcct.base.BaseFragment;
import com.example.dcct.bean.BackResultData;
import com.example.dcct.bean.LoginUserEntity;
import com.example.dcct.bean.TokenEntity;
import com.example.dcct.bean.UserEntity;
import com.example.dcct.databinding.FragmentLoginBinding;
import com.example.dcct.model.Impl.LoginModelImp;
import com.example.dcct.model.LoginModel;
import com.example.dcct.presenter.LoginPresenter;
import com.example.dcct.view.LoginCallback;
import com.google.android.material.textfield.TextInputLayout;

import java.util.regex.Matcher;
import java.util.regex.Pattern;
public class LoginFragment extends BaseFragment implements LoginCallback {

    private String email;
    private String password;
    private InformationDetermine mInformationDetermine;
    private LoginPresenter mLoginPresenter;
    private FragmentLoginBinding mBinding;

    public LoginFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        mBinding = FragmentLoginBinding.inflate( getLayoutInflater() );
        return mBinding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated( view, savedInstanceState );
        //检查输入邮箱格式
        checkEmailFormat( mBinding.textInputLayoutEmail.getEditText(),mBinding.textInputLayoutEmail );
        //点击登录按钮
        mBinding.ivLogin.setOnClickListener( v -> {
            if (mBinding.textInputLayoutEmail.getEditText() != null && mBinding.textInputLayoutPwd.getEditText() != null) {
                email = mBinding.textInputLayoutEmail.getEditText().getText().toString();
                password = mBinding.textInputLayoutPwd.getEditText().getText().toString();

                if (email.contentEquals( "" ) && !password.contentEquals( "" )) {
                    Toast.makeText( getActivity(), "邮箱号为空", Toast.LENGTH_SHORT ).show();
                } else if (!email.contentEquals( "" ) && password.contentEquals( "" )) {
                    Toast.makeText( getActivity(), "密码为空", Toast.LENGTH_SHORT ).show();
                } else if (email.contentEquals( "" ) && password.contentEquals( "" )) {
                    Toast.makeText( getActivity(), "邮箱、密码均为空！", Toast.LENGTH_SHORT ).show();
                } else {
                    //向服务器提交数据,将password进行MD5算法加密
                    LoginUserEntity loginUserEntity = new LoginUserEntity( email, MD5Decode32( password ) );
                    mLoginPresenter = new LoginPresenter();
                    mLoginPresenter.attachView( this );
                    mLoginPresenter.fetchUserEntity( loginUserEntity );
                    //在onLoadLoginData方法中接受数据
                }
            }
        } );
    }

    @Override
    public void onLoadLoginData(BackResultData<TokenEntity> backData) {
        if (backData.isState()) {
            if (getActivity() != null) {
                Toast.makeText( getActivity(), backData.getMsg(), Toast.LENGTH_SHORT ).show();
                //将登录成功的信息使用SharedPreferences存储
                SharedPreferences preferences = getActivity().getSharedPreferences( "SHARE_APP_LOGIN", Context.MODE_PRIVATE );
                preferences.edit().putBoolean( "LOGIN_SUCCESS", true ).apply();
            }
            TokenEntity dataBean = backData.getData();
            SharedPreferences preferences = getActivity().getSharedPreferences( "SHARE_APP_DATA", Context.MODE_PRIVATE );

//            Log.d( "uid",String.valueOf( dataBean.getUid() ) );
//            preferences.edit().putString( "nickname", dataBean.getNickname() )
//                    .putLong( "uid", dataBean.getUid() )
//                    .apply();
            //向父活动发送登录成功的通知
            mInformationDetermine.loginSuccess();
        } else {
            Toast.makeText( getActivity(), backData.getMsg(), Toast.LENGTH_SHORT ).show();
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        if (mLoginPresenter != null) {
            mLoginPresenter.detachView();
        }
    }

    @Override
    public void showErrorMsg(String msg) {
        Toast.makeText( getActivity(), msg, Toast.LENGTH_SHORT ).show();
    }

    public interface InformationDetermine {
        void loginSuccess();
    }

    // TODO: Rename method, update argument and hook method into UI event
    public void onButtonPressed() {
        if (mInformationDetermine != null) {
            mInformationDetermine.loginSuccess();
        }
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach( context );
        if (context instanceof InformationDetermine) {
            mInformationDetermine = (InformationDetermine) context;
        } else {
            throw new RuntimeException( context.toString()
                    + " must implement InformationDetermine" );
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mInformationDetermine = null;
    }
}

File: app/src/main/java/com/example/dcct/ui/fragment/LoginFragment.kt
package com.example.dcct.ui.fragment

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.example.dcct.base.BaseFragment
import com.example.dcct.bean.BackResultData
import com.example.dcct.bean.LoginUserEntity
import com.example.dcct.bean.UserEntity
import com.example.dcct.databinding.FragmentLoginBinding
import com.example.dcct.presenter.LoginPresenter
import com.example.dcct.view.LoginCallback

class LoginFragment : BaseFragment(), LoginCallback {
    private var email: String? = null
    private var password: String? = null
    private var mInformationDetermine: InformationDetermine? = null
    private var mLoginPresenter: LoginPresenter? = null
    private var mBinding: FragmentLoginBinding? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        mBinding = FragmentLoginBinding.inflate(layoutInflater)
        return mBinding!!.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        //检查输入邮箱格式
        checkEmailFormat(mBinding!!.textInputLayoutEmail.editText, mBinding!!.textInputLayoutEmail)
        //点击登录按钮
        mBinding!!.ivLogin.setOnClickListener { v: View? ->
            if (mBinding!!.textInputLayoutEmail.editText != null && mBinding!!.textInputLayoutPwd.editText != null) {
                email = mBinding!!.textInputLayoutEmail.editText!!.text.toString()
                password = mBinding!!.textInputLayoutPwd.editText!!.text.toString()
                if (email!!.contentEquals("") && !password!!.contentEquals("")) {
                    Toast.makeText(activity, "邮箱号为空", Toast.LENGTH_SHORT).show()
                } else if (!email!!.contentEquals("") && password!!.contentEquals("")) {
                    Toast.makeText(activity, "密码为空", Toast.LENGTH_SHORT).show()
                } else if (email!!.contentEquals("") && password!!.contentEquals("")) {
                    Toast.makeText(activity, "邮箱、密码均为空！", Toast.LENGTH_SHORT).show()
                } else {
                    //向服务器提交数据,将password进行MD5算法加密
//                    LoginUserEntity loginUserEntity = new LoginUserEntity( email, MD5Decode32( password ) );
                    val loginUserEntity = LoginUserEntity(email!!, password!!)
                    mLoginPresenter = LoginPresenter()
                    mLoginPresenter!!.attachView(this)
                    mLoginPresenter!!.fetchUserEntity(loginUserEntity)
                    //在onLoadLoginData方法中接受数据
                }
            }
        }
    }

    override fun onLoadLoginData(backData: BackResultData<UserEntity>) {
        if (backData.isState && activity != null) {
            Toast.makeText(activity, backData.msg, Toast.LENGTH_SHORT).show()
            //将登录成功的信息使用SharedPreferences存储
            var preferences = requireActivity().getSharedPreferences("SHARE_APP_LOGIN", Context.MODE_PRIVATE)
            preferences.edit().putBoolean("LOGIN_SUCCESS", true).apply()
            preferences = requireActivity().getSharedPreferences("SHARE_APP_DATA", Context.MODE_PRIVATE)
            preferences.edit().putString("nickname", backData.data!!.nickname)
                    .putLong("uid", backData.data!!.uid.toLong())
                    .apply()
            //向父活动发送登录成功的通知
            mInformationDetermine!!.loginSuccess()
        } else {
            Toast.makeText(activity, backData.msg, Toast.LENGTH_SHORT).show()
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        mBinding = null
        if (mLoginPresenter != null) {
            mLoginPresenter!!.detachView()
        }
    }

    override fun showErrorMsg(msg: String?) {
        Toast.makeText(activity, msg, Toast.LENGTH_SHORT).show()
    }

    interface InformationDetermine {
        fun loginSuccess()
    }

    // TODO: Rename method, update argument and hook method into UI event
    fun onButtonPressed() {
        if (mInformationDetermine != null) {
            mInformationDetermine!!.loginSuccess()
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mInformationDetermine = if (context is InformationDetermine) {
            context
        } else {
            throw RuntimeException(context.toString()
                    + " must implement InformationDetermine")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mInformationDetermine = null
    }
}